File: easy/cpp/0121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() <= 1) return 0;
        int Min = prices[0];
        int ret = 0;
        for(int i = 1 ; i < prices.size(); i++){
            if(prices[i] - Min > ret)ret = prices[i] - Min;
            if(prices[i] < Min) Min = prices[i];
        }
        return ret;
    }
};

File: easy/python/0121.py
class Solution:
    def maxProfit(self, prices: 'List[int]') -> 'int':
        if not prices:
            return 0
        buy = prices[0]
        res = 0
        for price in prices:
            res = max([res, price - buy])
            buy = min([price, buy])
        return res